File: Javascript/bubblesort.js
const bubbleSort = (arr) => {
    let aux, 
        sorted = false
    while(!sorted){
        sorted = true
        for(let i = 0; i < arr.length; i++){
            if(arr[i + 1] && arr[i] > arr[i + 1]){
               sorted         = false 
               aux            = arr[i]
               arr[i]         = arr[i + 1]
               arr[i + 1]     = aux 
            }
        }
    }
    return arr
}

console.log(bubbleSort([64,321,47,21,92,11,3]))

File: Ruby/bubbleSort.rb
def bubble_sort(array)
  sorted = false

  until sorted
    sorted = true

    (1...array.length).each do |i|
      if array[i - 1] > array[i]
        sorted = false
        array[i - 1], array[i] = array[i], array[i - 1]
      end
    end
  end

  array
end
